pyvrp/cpp/search/SwapStar.cpp

```diff
@@ -168,7 +170,7 @@ Cost SwapStar::evaluateMove(Route::Node *U,
 
     deltaCost -= costEvaluator.twPenalty(route->timeWarp());
 
-    auto const loadDiff = data.client(best.U->client()).demand
+    auto const loadDiff = data.client(U->client()).demand
                           - data.client(remove->client()).demand;
```

Comment:
This was the issue. Introduced in #302, happens so extremely rarely it's no wonder it was that hard to reproduce. I'll write a test exercising this in a minute.